subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -395,6 +372,61 @@ u8_t ll_version_ind_send(u16_t handle)
 	return 0;
 }
 
+#if defined(CONFIG_BT_CTLR_DATA_LENGTH)
+u32_t ll_length_req_send(u16_t handle, u16_t tx_octets, u16_t tx_time)
+{
+	struct ll_conn *conn;
+
+	conn = ll_connected_get(handle);
+	if (!conn) {
+		return BT_HCI_ERR_UNKNOWN_CONN_ID;
+	}
+
+	if ((conn->llcp_req != conn->llcp_ack) ||
+	    (conn->llcp_length.req != conn->llcp_length.ack)) {
+		return BT_HCI_ERR_CMD_DISALLOWED;
+	}
+
+	/* TODO: parameter check tx_octets and tx_time */
+
+	conn->llcp_length.state = LLCP_LENGTH_STATE_REQ;
+	conn->llcp_length.tx_octets = tx_octets;
+
+#if defined(CONFIG_BT_CTLR_PHY)
+	conn->llcp_length.tx_time = tx_time;
+#endif /* CONFIG_BT_CTLR_PHY */
+
+	conn->llcp_length.req++;
+
+	return 0;
+}
+
+void ll_length_default_get(u16_t *max_tx_octets, u16_t *max_tx_time)
+{
+	*max_tx_octets = default_tx_octets;
+	*max_tx_time = default_tx_time;
+}
+
+u32_t ll_length_default_set(u16_t max_tx_octets, u16_t max_tx_time)
+{
+	/* TODO: parameter check (for BT 5.0 compliance) */
+
+	default_tx_octets = max_tx_octets;
+	default_tx_time = max_tx_time;
+
+	return 0;
+}
+
+void ll_length_max_get(u16_t *max_tx_octets, u16_t *max_tx_time,
+		       u16_t *max_rx_octets, u16_t *max_rx_time)
+{
+	*max_tx_octets = LL_LENGTH_OCTETS_RX_MAX;
+	*max_tx_time = PKT_US(LL_LENGTH_OCTETS_RX_MAX, BIT(2));
+	*max_rx_octets = LL_LENGTH_OCTETS_RX_MAX;
+	*max_rx_time = PKT_US(LL_LENGTH_OCTETS_RX_MAX, BIT(2));
```

Comment:
BIT(2) means coded, which of course is a safe, but not very tight, bound. 
So this should be `#ifdef`'ed on whether coded phy is supported.